redis/connection.py

```diff
@@ -1085,49 +1095,96 @@ def __eq__(self, other):
 
     def reset(self):
         self.pid = os.getpid()
```

Comment:
Apparently, this code still exhibits the race condition described in https://github.com/andymccurdy/redis-py/issues/1138#issuecomment-512813988 and addressed in PR https://github.com/andymccurdy/redis-py/pull/1262.  

The PR does not change the behavior (w.r.t. issue #1138) for me. Moving this line to the end of the method does help.

Reply:
@gmbnomis Do you have a brief code sample that can repro what you're seeing?

Reply:
I see what's going on now. You're right, setting the PID should happen as the very last call as the PID check isn't protected via a lock for performance reasons.

Reply:
@gmbnomis I pushed the change to this PR. Would you mind testing this branch in your setup and see if it fixes your issue? I'd like to get this out in the next release.

Reply:
@andymccurdy I gave it a spin and I can't reproduce the problem anymore. Thank you for fixing the issue!

Reply:
Excellent! Thanks for helping to figure out what was happening and the testing effort! I'm going to merge this and release redis-py 3.4.0 now.